query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -262,6 +286,82 @@ private String convertOperator2String(Operator operator) {
     }
   }
 
+  private String handleFilterForMapAttribute(Filter filter, Builder paramsBuilder) {
+
+    if (!SUPPORTED_OPERATORS_FOR_MAP_ATTRIBUTES_WITH_SUBPATH.contains(filter.getOperator())) {
+      throw new UnsupportedOperationException(
+          "Unknown operator for map attributes:" + filter.getOperator());
+    }
+
+    LiteralConstant[] kvp = convertExpressionToMapLiterals(filter.getRhs(), filter.getLhs());
+    String keyCol = convertExpressionToMapKeyColumn(filter.getLhs());
+    String valCol = convertExpressionToMapValueColumn(filter.getLhs());
+
+    return keyCol
+        + " = "
+        + convertLiteralToString(kvp[MAP_KEY_INDEX], paramsBuilder)
+        + " AND "
+        + valCol
+        + " "
+        + convertOperator2String(filter.getOperator())
+        + " "
+        + convertLiteralToString(kvp[MAP_VALUE_INDEX], paramsBuilder)
+        + " AND "
+        + MAP_VALUE
+        + "("
+        + keyCol
+        + ","
+        + convertLiteralToString(kvp[MAP_KEY_INDEX], paramsBuilder)
+        + ","
+        + valCol
+        + ")"
+        + " "
+        + convertOperator2String(filter.getOperator())
+        + " "
+        + convertLiteralToString(kvp[MAP_VALUE_INDEX], paramsBuilder);
+  }
+
+  private boolean isGroupingSelectionForMapAttribute(
```

Comment:
is this method needed (given that convertExpression2StringForMapAttribute is falling back to convertExpression2String)?

Reply:
It is falling back in a subset of cases only. For eg: when this function returns true, it will not fallback.

Reply:
Ah, I see the difference in the conditional check in this method and `convertExpression2StringForMapAttribute`, I think a cleaner way would be to assume that `convertExpression2StringForMapAttribute` is called with the correct input and it doesn't fall back?

Reply:
resolved